### PR TITLE
feat(ci.jenkins.io) cleans up plugin list

### PR DESCRIPTION
**Pull request by dduportal**
This PR is an attempt at cleaning up the plugins installed on ci.jenkins.io by the puppet infrastructure as code system **when the controller is running**.

- Comment added for each plugins present in the list that we can justify
- Removed a set of unused/should not use plugins

Please not this list is only a "install them if specified". It's not exhaustive and does not map 1:1 to the reality of ci.jenkins.io (at least not until https://github.com/jenkins-infra/helpdesk/issues/3070).

### CODE REVIEW

File: hieradata/clients/azure.ci.jenkins.io.yaml [hunk 1]
```diff
@@ -431,45 +431,38 @@ profile::jenkinscontroller::jcasc:
...
+  - github # Provides basic GitHub integration for jobs
+  - github-checks # Allows GitHub Jobs to create checks in GitHub API
+  - github-branch-source # Provides Job integration to get code from GitHub
+  - generic-tool # Ligthweight tool definition (should replace 'groovy' plugin and tools)
+  - groovy # TODO: remove in favor of generic-tool
+  - kubernetes # Kubernetes Container Agents
+  - jobConfigHistory # TODO: remove in favor of a configuration as code job definition (JobDSL?)
+  - junit-realtime-test-reporter # Provides the 'realTimeJunit' keyword used by the ATH - https://github.com/jenkinsci/acceptance-test-harness/blob/1cfec809d04b2a098f769ff662320d0e873dfb9f/Jenkinsfile#L60
+  - ldap # Required for authentication/authorization
+  - lockable-resources # Provides the pipelie keyword 'lock'
```
Comment (timja) [inline hieradata/clients/azure.ci.jenkins.io.yaml]:
```suggestion
  - lockable-resources # Provides the pipeline keyword 'lock'
```

**Review by timja — APPROVED** [review]

**Comment by dduportal** [comment]
~Putting as drafty until the infra team puts a schedule + communication plan on this one~

Missing time on the infra weekly meeting. Gotta ask reviewers asynchronously.

**Review by MarkEWaite — APPROVED** [review]
I compared line by line the "before" and "after" and agree that this is a good choice.

Will the removal of pipeline-utility-steps break any known jobs on ci.jenkins.io?

File: hieradata/clients/azure.ci.jenkins.io.yaml [hunk 2]
```diff
@@ -431,45 +431,38 @@ profile::jenkinscontroller::jcasc:
...
+  - credentials-binding # Bind Jenbkins credentials to variables in jobs
+  - docker-workflow # Provides the "withDockerContainer" pipelie keyword
+  - ec2 # AWS EC2 VM agents
+  - embeddable-build-status # https://github.com/jenkins-infra/helpdesk/issues/3013
+  - git # Used for... git
+  - github # Provides basic GitHub integration for jobs
+  - github-checks # Allows GitHub Jobs to create checks in GitHub API
+  - github-branch-source # Provides Job integration to get code from GitHub
+  - generic-tool # Ligthweight tool definition (should replace 'groovy' plugin and tools)
+  - groovy # TODO: remove in favor of generic-tool
```
Comment (daniel-beck) [inline hieradata/clients/azure.ci.jenkins.io.yaml:453]: ❤️

**Comment by dduportal** [comment]
> I compared line by line the "before" and "after" and agree that this is a good choice.
> 
> Will the removal of pipeline-utility-steps break any known jobs on ci.jenkins.io?

It might, but identifying these jobs and fixing them is clearly more important than breaking them. Rationale is that most of these keywords can easily be replaced by an `sh ''` instruction to defer their execution to the agents.

Two cases were broken when we temporarly uninstalled the plugin last week:

- @halkeye already published a pipeline-library for the "deployment records" of the static websites, which was the most annoying (and easier to catch because breaking PR preview for websites).
- Publication of the results for the pipeline generator jobs (that you fixed already ❤️ ).

**Comment by dduportal** [comment]
I'm proceeding to merge this one, as 3 "non-usual infra team members" approval were done. Thanks a lot!

**Comment by dduportal** [comment]
For information, plugins were "cleaned up" as part of the LTS update to 2.361.4.

**Pull request merged**